sw/device/silicon_creator/rom/e2e/boot_policy_flash_ecc_error/flash_ecc_error_test.c

```diff
@@ -0,0 +1,140 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/base/chip.h"
+
+#include "flash_ctrl_regs.h"  // Generated.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"  // Generated.
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_flash_ctrl_state_t flash_ctrl;
+static dif_otp_ctrl_t otp_ctrl;
+
+static dif_flash_ctrl_region_properties_t kFlashFullAccessScrambledEcc = {
+    .ecc_en = kMultiBitBool4True,
+    .high_endurance_en = kMultiBitBool4False,
+    .scramble_en = kMultiBitBool4True,
+    .erase_en = kMultiBitBool4True,
+    .prog_en = kMultiBitBool4True,
+    .rd_en = kMultiBitBool4True};
+
+/**
+ * ROM_EXT slots.
+ */
+typedef enum rom_ext_slot {
+  kRomExtSlotA = 0,
+  kRomExtSlotB = 1,
+} rom_ext_slot_t;
+
+enum {
+  // Manifest Offsets; See `sw/device/silicon_creator/lib/manifest.h`.
+  kManifestIdOffest = 820,
```

Comment:
nit: Consider adding an assertion to ensure this offset is always aligned.

Reply:
Ah yeah, one exists here: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/manifest.h;drc=30b710a40e6a68786e6436e285ab8294104374ca;l=342

But I will add a duplicate here in my follow up PR that adds the example prototype and test enhancements so it is more obvious what code needs to be updated if this ever changes.